pkg/gateway/gateway.go

```diff
@@ -186,11 +186,11 @@ func (n *jfsObjects) tpath(p ...string) string {
 }
 
 func (n *jfsObjects) upath(bucket, uploadID string) string {
-	return n.tpath(bucket, "uploads", uploadID)
+	return n.tpath(bucket, "uploads", uploadID[:2], uploadID)
 }
 
 func (n *jfsObjects) ppath(bucket, uploadID, part string) string {
-	return n.tpath(bucket, "uploads", uploadID, part)
+	return n.tpath(bucket, "uploads", uploadID[:2], uploadID, part)
```

Comment:
this is not compatible to previous version, how can we provide a smooth upgrade?

Reply:
I suppose there are no compatibility issues. After upgrade, new objects will be written to the new path, and old objects will be cleaned up by `cleanupDir`.

Reply:
The running multipart-upload will fail after upgrade

Reply:
Do you mean uploading some parts using the old path and some parts using the new path? In this case, the `CompleteMultipartUpload` call will fail, and the user will get an error. Most tools will retry after receiving a failure.

Reply:
Commonly rclone will retry the entire upload if `CompleteMultipartUpload` fails multiple times.

For other tools, `CompleteMultipartUpload` will eventually fail, and the user will be notified. I suppose it is acceptable to encounter several failed requests during an upgrade.

Reply:
It's better to make the upgrade smooth if possible

Reply:
Do you have any suggestions?

Reply:
we can fallback to original path when 1) find uploaded part 2) list parts